src/plopp/backends/pythreejs/mesh3d.py

```
@@ -0,0 +1,179 @@
+                    (self._data.coords["x"].shape[0], 3),
+                ).astype('float32')
+            ),
+        }
+
+        self.geometry = p3.BufferGeometry(index=index, attributes=attributes)
+        self.material = p3.MeshBasicMaterial(
+            vertexColors='VertexColors',
+            transparent=True,
+            side='DoubleSide',
+            opacity=opacity,
+            depthTest=opacity > 0.5,
+        )
+        self.mesh = p3.Mesh(geometry=self.geometry, material=self.material)
+        self.edges = (
+            p3.LineSegments(
+                p3.EdgesGeometry(self.geometry),
+                p3.LineBasicMaterial(
+                    color=edgecolor or 'black',
+                    linewidth=2,
+                    opacity=opacity,
+                    transparent=True,
+                ),
+            )
+            if edgecolor is not None
+            else None
+        )
+        self._canvas.add(self.mesh)
+        if self.edges is not None:
+            self._canvas.add(self.edges)
+
+    def set_colors(self, rgba):
+        """
+        Set the mesh's rgba colors:
+
+        Parameters
+        ----------
+        rgba:
+            The array of rgba colors.
+        """
+        self.geometry.attributes["color"].array = rgba[..., :3].astype(
+            'float32', copy=False
+        )
+
+    def update(self, new_values):
+        """
+        Update mesh array with new values.
+
+        Parameters
+        ----------
+        new_values:
+            New data to update the mesh values from.
+        """
+        self._data = new_values
+        # TODO: for now we only update the colors of the mesh. Updating the positions
+        # of the vertices is doable but is made more complicated by the edges geometry,
+        # whose positions cannot just be updated. A new geometry and edge lines would
+        # have to be created, the old one removed from the scene and the new one added.
```

It doesn't looks like `update` changes the colours to me. That is what `set_colors` does. So could `update` simply raise an exception instead.
As it stands, this could hide errors.

---

It's true that the colors are not changed. However, the values of the `._data` should be updated, because this is what the `colormapper` uses to determine the colorbar range. So it should not raise, but I will update the comment.